lab1/src/icecream.ts

```diff
@@ -0,0 +1,117 @@
+enum Size {
+    Small = "small",
+    Large = "large"
+}
+
+enum Topping {
+    Chocolate = "chocolate",
+    Caramel = "caramel",
+    Berries = "berries"
+}
+
+enum YesNo {
+    Yes = "yes",
+    No = "no"
+}
+
+const SizeCosts = {
+    [Size.Small]: 10,
+    [Size.Large]: 25
+}
+
+const ToppingCosts = {
+    [Topping.Chocolate]: 5,
+    [Topping.Caramel]: 6,
+    [Topping.Berries]: 10
+}
+
+const MarshmallowCost = 5;
+
+function calculateIceCreamCost(size: Size, toppings: Topping[], addMarshmallow: YesNo): number {
+    let cost = SizeCosts[size];
+
+    toppings.forEach((topping) => {
+        cost += ToppingCosts[topping];
+    });
+
+    if (addMarshmallow === YesNo.Yes) {
+        cost += MarshmallowCost;
+    }
+
+    return cost;
+}
+
+function isSize(input: string): input is Size {
```

Comment:
functions isSize, isTopping, isYesNo do th same thing, instead cereate helper function.

Reply:
created isValidEnumValue to assert the values and validateEnumValue, validateEnumArray to validate input